project-addons/sale_display_stock/product.py

```diff
@@ -45,16 +45,20 @@ def _get_input_loc_stock(self):
 
     @api.one
     def _get_in_production_stock(self):
-        moves = self.env["stock.move"].search([('product_id', '=',
-                                                self.product_variant_ids[0].id),
-                                               ('purchase_line_id', '!=',
-                                                False),
-                                               ('picking_id', '=', False),
-                                               ('state', '!=', 'cancel')])
-        qty = 0.0
-        for move in moves:
-            qty += move.product_uom_qty
-        self.qty_in_production = qty
+        if self.product_variant_ids:
+            moves = self.env["stock.move"].search([('product_id', '=',
+                                                    self.product_variant_ids.id),
```

Comment:
El operador sería 'in' no '=' y sería self.product_variant_ids.ids no self.product_variant_ids.id